internal/armada/repository/usage.go

```diff
@@ -161,6 +167,41 @@ func (r *RedisUsageRepository) UpdateClusterLeased(report *api.ClusterLeasedRepo
 	return err
 }
 
+func (r *RedisUsageRepository) GetClusterQueueResourceUsage() (map[string]*schedulerobjects.ClusterResourceUsageReport, error) {
+	result, err := r.db.HGetAll(clusterQueueResourceUsageKey).Result()
+	if err != nil {
+		return nil, fmt.Errorf("[RedisUsageRepository.GetClusterQueueResourceUsage] error reading from database: %s", err)
+	}
+	reports := make(map[string]*schedulerobjects.ClusterResourceUsageReport)
+
+	for k, v := range result {
+		report := &schedulerobjects.ClusterResourceUsageReport{}
+		err = proto.Unmarshal([]byte(v), report)
+		if err != nil {
+			return nil, fmt.Errorf("[RedisUsageRepository.GetClusterQueueResourceUsage] error unmarshalling: %s", err)
+		}
+		reports[k] = report
+	}
+	return reports, nil
+}
+
+func (r *RedisUsageRepository) UpdateClusterQueueResourceUsage(cluster string, resourceUsage *schedulerobjects.ClusterResourceUsageReport) error {
+
+	data, err := proto.Marshal(resourceUsage)
+	if err != nil {
+		return fmt.Errorf("[RedisUsageRepository.UpdateClusterQueueResourceUsage] error marshalling report: %s", err)
```

Comment:
We should use `errors.Errorf`. Then we don't need the `[]` part either.

Reply:
Ideally yes but the rest of the  file is using this format and I didn't want to make this change as part of this pr
I think it would be better to have a separate pr where all that's done is to change to the new way of doing errors.